src/main/java/uk/nhs/prm/e2etests/repository/CoreRepository.java

```diff
@@ -35,11 +41,51 @@ private Optional<CoreRecord> getCoreByInboundConversationId(String inboundConver
         return Optional.ofNullable(table.getItem(key));
     }
 
+    private PageIterable<CoreRecord> getFragmentsByMessageId(String inboundConversationId, String fragmentMessageId) {
+        //find items with inbound ID and FRAGMENT at start of Layer
+        final QueryConditional condition = QueryConditional.keyEqualTo(k -> k.partitionValue(inboundConversationId));
+        final Expression filter = Expression.builder()
+                .expression("InboundMessageId = :id")
+                .expressionValues(Map.of(":id", AttributeValue.builder().s(fragmentMessageId).build()))
+                .build();
+
+        final QueryEnhancedRequest request = QueryEnhancedRequest.builder()
+                .queryConditional(condition)
+                .filterExpression(filter)
+                .build();
+
+        return table.query(request);
```

Comment:
Can we have this return `table.query(request).items();` so that we're not returning a `PageIterable<CoreRecord>` which is always going to have to have `.items()` added after it before we can do anything with it?